File: spiritual_root-common/src/main/java/io/github/solusmods/eternalcore/spiritual_root/impl/network/SpiritualRootNetwork.java
package io.github.solusmods.eternalcore.spiritual_root.impl.network;

import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;
import io.github.solusmods.eternalcore.spiritual_root.impl.network.c2s.RequestSpiritualRootAdvancePacket;

public class SpiritualRootNetwork {

    private SpiritualRootNetwork() {
    }

    public static void init() {
        NetworkUtils.registerC2SPayload(RequestSpiritualRootAdvancePacket.TYPE,
                RequestSpiritualRootAdvancePacket.STREAM_CODEC, RequestSpiritualRootAdvancePacket::handle);
    }


}

File: spiritual_root-common/src/main/kotlin/io/github/solusmods/eternalcore/spiritual_root/impl/network/SpiritualRootNetwork.kt
package io.github.solusmods.eternalcore.spiritual_root.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils.registerC2SPayload
import io.github.solusmods.eternalcore.spiritual_root.impl.network.c2s.RequestSpiritualRootAdvancePacket
import io.github.solusmods.eternalcore.spiritual_root.impl.network.c2s.SyncSpiritualRootStoragePayload

object SpiritualRootNetwork {
    fun init() {
        registerC2SPayload<RequestSpiritualRootAdvancePacket?>(
            RequestSpiritualRootAdvancePacket.Companion.TYPE,
            RequestSpiritualRootAdvancePacket.Companion.STREAM_CODEC
        ) { obj: RequestSpiritualRootAdvancePacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        registerC2SPayload<SyncSpiritualRootStoragePayload?>(
            SyncSpiritualRootStoragePayload.Companion.TYPE,
            SyncSpiritualRootStoragePayload.Companion.STREAM_CODEC
        ) { obj: SyncSpiritualRootStoragePayload?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
    }
}
